src/atdf/patch.rs

```diff
@@ -52,6 +52,14 @@ pub fn signals_to_port_fields(chip: &mut chip::Chip, tree: &xmltree::Element) ->
     Ok(())
 }
 
+fn correct_reg_name(reg_name: &String, port_name: char) -> bool {
```

Comment:
In Rust, you should pass a `&str` when you have a reference to a string.  Using `&String` is rarely the right thing.

```suggestion
fn correct_reg_name(reg_name: &str, port_name: char) -> bool {
```